service-front/app/src/Actor/src/Handler/RequestActivationKey/ReferenceNumberHandler.php

```diff
@@ -23,22 +24,25 @@ class ReferenceNumberHandler extends AbstractRequestKeyHandler implements UserAw
 {
     private RequestReferenceNumber $form;
     private RemoveAccessForAllSessionValues $removeAccessForAllSessionValues;
+    private FeatureEnabled $featureEnabled;
 
 
     public function __construct(
         TemplateRendererInterface $renderer,
         AuthenticationInterface $authenticator,
         UrlHelper $urlHelper,
         LoggerInterface $logger,
-        RemoveAccessForAllSessionValues $removeAccessForAllSessionValues
+        RemoveAccessForAllSessionValues $removeAccessForAllSessionValues,
+        FeatureEnabled $featureEnabled
     ) {
         parent::__construct($renderer, $authenticator, $urlHelper, $logger);
         $this->removeAccessForAllSessionValues = $removeAccessForAllSessionValues;
+        $this->featureEnabled = $featureEnabled;
     }
 
     public function handle(ServerRequestInterface $request): ResponseInterface
     {
-        $this->form = new RequestReferenceNumber($this->getCsrfGuard($request));
+        $this->form = new RequestReferenceNumber($this->getCsrfGuard($request), $this->featureEnabled);
```

Comment:
I think, in this case, it would be better to pass the value of the flag to the form?

```suggestion
        $this->form = new RequestReferenceNumber($this->getCsrfGuard($request), ($this->featureEnabled)('allow_meris_lpas'));
```

The form doesn't need to know about all feature flags. Just the one that affects it. In fact, you could make the same argument for the handler but I can see that a handler might need multiple flag checks.